frontend/projects/upgrade/src/app/core/feature-flags/store/feature-flags.effects.ts

```diff
@@ -177,19 +179,81 @@ export class FeatureFlagsEffects {
 
   fetchFeatureFlagById$ = createEffect(() =>
     this.actions$.pipe(
-      ofType(FeatureFlagsActions.actionFetchFeatureFlagById),
+      ofType(featureFlagsActions.actionFetchFeatureFlagById),
       map((action) => action.featureFlagId),
       filter((featureFlagId) => !!featureFlagId),
       switchMap((featureFlagId) =>
         this.featureFlagsDataService.fetchFeatureFlagById(featureFlagId).pipe(
           map((data: FeatureFlag) => {
-            return FeatureFlagsActions.actionFetchFeatureFlagByIdSuccess({ flag: data });
+            return featureFlagsActions.actionFetchFeatureFlagByIdSuccess({ flag: data });
           }),
-          catchError(() => [FeatureFlagsActions.actionFetchFeatureFlagByIdFailure()])
+          catchError(() => [featureFlagsActions.actionFetchFeatureFlagByIdFailure()])
         )
       )
     )
   );
 
+  emailFeatureFlagData$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(featureFlagsActions.actionEmailFeatureFlagData),
+      map((action) => ({ featureFlagId: action.featureFlagId })),
+      withLatestFrom(this.store$.pipe(select(selectCurrentUser))),
+      filter(([{ featureFlagId }, { email }]) => !!featureFlagId && !!email),
+      switchMap(([{ featureFlagId }, { email }]) =>
+        this.featureFlagsDataService.emailFeatureFlagData(featureFlagId, email).pipe(
+          tap(() => {
+            email
+              ? this.notificationService.showSuccess(`Email will be sent to ${email}`)
+              : this.notificationService.showSuccess('Email will be sent to registered email');
```

Comment:
well, we're sending a success message here before we know if it's successful, we should do it in the `success` map, no `tap` necessary as far as I can tell?

Reply:
the `tap` seems unnecesary, that call can just go in the `map` with the rest of the success handling?

Reply:
there was another part to remove, the second part of this ternary is dead code, it can't get there because `email` will always be truthy due to the filter a little above.

```
email
              ? this.notificationService.showSuccess(`Email will be sent to ${email}`)
              : this.notificationService.showSuccess('Email will be sent to registered email'
```

can just be:

```
this.notificationService.showSuccess(`Email will be sent to ${email}
```